rpcdetect/rest.go

```diff
@@ -0,0 +1,97 @@
+package rpcdetect
+
+import (
+	"fmt"
+	"net/http"
+	"slices"
+
+	"google.golang.org/protobuf/proto"
+
+	"github.com/pokt-network/shannon-sdk/types"
+)
+
+var (
+	// defaultRESTErrorReply is the default REST error reply to be sent if the
+	// formatRESTError function fails to format the appropriate one.
+	defaultRESTErrorReply   *types.POKTHTTPResponse
+	defaultRESTErrorReplyBz []byte
+)
+
+// init initializes the default REST error reply. This function is called before
+// the main function and panics if it fails to marshal the default REST error reply,
+// making the program exit early.
+func init() {
+	// Initialize the default REST error reply
+	header := &types.Header{
+		Key:    contentTypeHeaderKey,
+		Values: []string{"text/plain"},
+	}
+	headers := map[string]*types.Header{contentTypeHeaderKey: header}
+
+	defaultRESTErrorReply = &types.POKTHTTPResponse{
+		StatusCode: http.StatusInternalServerError,
+		Header:     headers,
+		BodyBz:     []byte(`Internal error`),
+	}
+
+	var err error
+	defaultRESTErrorReplyBz, err = proto.Marshal(defaultRESTErrorReply)
+	if err != nil {
+		panic(err)
+	}
+}
+
+// isREST checks if the given POKTHTTPRequest is a REST request.
+func isREST(poktRequest *types.POKTHTTPRequest) bool {
```

Comment:
I feel like almost all the functions take a req as a param field.

Given that goland has receivers and an "object oriented approach", it's more appropriate than going down this function path.